game/pokemon.go

```diff
@@ -1,10 +1,118 @@
+    p.allCharacters = make([]PokemonCharacter, 6)
+    names := []string{"Pikachu", "Blossomite", "Charmander", "Ditto", "Eevee", "Flareon"}
+    // setup characters
+    for i := 0; i < 6; i++ {
+        p.allCharacters[i] = PokemonCharacter{Name: names[i], strength: p.getRandomStrength(80, 300)}
+    }
+    // ask user to pick one
+    fmt.Println("Select the Pokemon Character You Want: ")
+    fmt.Println("----------")
+    fmt.Println("Characters:")
+    fmt.Println("----------")
+    for i, name := range names {
+        fmt.Printf("     %d. %s \n", i+1, name)
+    }
+    selectedIndex := -1
+    // get user selection from scan
+    _, err := fmt.Scan(&selectedIndex)
+    if err != nil {
+        fmt.Println("The selection must be a number representing the character")
+        return
+    }
+    if selectedIndex <= 0 || selectedIndex >= len(p.allCharacters) {
+        fmt.Println("The selection is not valid")
+        return
+    }
+    selectedIndex = selectedIndex - 1
+    p.userSelection = p.allCharacters[selectedIndex]
+
+    for round := 1; round <= numRounds; round++ {
+        // select 3 random fighters
+        // find 3 number between 0 and 5 that is not equal to the selected index and no duplicates
+        number1 := getRandomNumber(5, selectedIndex)
+        number2 := getRandomNumber(5, selectedIndex, number1)
+        number3 := getRandomNumber(5, selectedIndex, number1, number2)
+
+        fighter1 := p.allCharacters[number1]
+        fighter2 := p.allCharacters[number2]
+        fighter3 := p.allCharacters[number3]
+
+        fmt.Printf("Round %d: Your Pokemon '%s' (strength = %d); Fights '%s' (strength = '%d')\n", round, p.userSelection.Name, p.userSelection.strength, fighter1.Name, fighter1.strength)
+        winner, isDraw := p.fight(p.userSelection, fighter1)
+        if winner != p.userSelection {
+            fmt.Printf("Sorry you lost the fight to: %s\n", winner.Name)
+            continue
+        }
+        if !isDraw {
+            fmt.Println("You won Round 1")
+        }
+
+        fmt.Printf("Round %d: Your Pokemon '%s' (strength = %d); Fights '%s' (strength = '%d')\n", round, p.userSelection.Name, p.userSelection.strength, fighter2.Name, fighter2.strength)
+        winner, isDraw = p.fight(p.userSelection, fighter2)
+        if winner != p.userSelection {
+            fmt.Printf("Sorry you lost the fight to: %s\n", winner.Name)
+            continue
+        }
+        if !isDraw {
+            fmt.Println("You won Round 2")
+        }
+
+        fmt.Printf("Round %d: Your Pokemon '%s' (strength = %d); Fights '%s' (strength = '%d')\n", round, p.userSelection.Name, p.userSelection.strength, fighter3.Name, fighter3.strength)
+        winner, isDraw = p.fight(p.userSelection, fighter3)
+        if winner != p.userSelection {
+            fmt.Printf("Sorry you lost the fight to: %s\n", winner.Name)
+            continue
+        }
+        fmt.Println("Congratulations, you won all 3 rounds")
```

Comment:
This should only print on the last round and should indicate the total number of rounds and not always 3